### PR TITLE
Fixed build with XCode on MacOS

### PR DESCRIPTION
**Pull request by noseglasses**
To build Arduino-CMake with XCode on MacOS some linker flags have to be changed that differ between gcc and clang.

Also it turned out that there was a missing target dependency that only applies when building shared libraries with Arduino-CMake.

### CODE REVIEW

File: cmake/Platform/Core/LibraryFinder.cmake [hunk 1]
```diff
@@ -144,6 +144,6 @@ function(find_arduino_libraries VAR_NAME SRCS ARDLIBS)
     endif ()
 
     _REMOVE_BLACKLISTED_LIBRARIES("${ARDUINO_LIBS}" FILTERED_LIBRARIES)
-    set(${VAR_NAME} ${FILTERED_LIBRARIES} PARENT_SCOPE)
+    set(${VAR_NAME} "${FILTERED_LIBRARIES}" PARENT_SCOPE)
```
Comment (noseglasses) [inline cmake/Platform/Core/LibraryFinder.cmake:147]: It is always better to quote lists when passing them as they might contain spaces.

File: cmake/Platform/Core/Targets/ArduinoFirmwareTargetCreator.cmake [hunk 2]
```diff
@@ -39,7 +39,20 @@ function(create_arduino_firmware_target TARGET_NAME BOARD_ID ALL_SRCS ALL_LIBS
       # all symbols from the intermediate static libraries end up in the
       # static library
       #
-      target_link_libraries(${TARGET_NAME} PUBLIC "-Wl,--whole-archive" ${ALL_LIBS} "-Wl,--no-whole-archive")
+      if(CMAKE_HOST_APPLE)
+         foreach(lib ${ALL_LIBS})
+            if(NOT "${linker_flags}" STREQUAL "")
+               set(linker_flags "${linker_flags} ")
+            endif()
+            set(linker_flags "${linker_flags}-Wl,-force_load,lib${lib}.a")
```
Comment (noseglasses) [inline cmake/Platform/Core/Targets/ArduinoFirmwareTargetCreator.cmake:47]: Clang's `-force_load` is the equivalent to gcc's `--whole-archive`, although used in a slightly different way.

File: cmake/Platform/Core/Targets/ArduinoFirmwareTargetCreator.cmake [hunk 3]
```diff
@@ -39,7 +39,20 @@ function(create_arduino_firmware_target TARGET_NAME BOARD_ID ALL_SRCS ALL_LIBS
...
+         endforeach()
+#          message("linker_flags: ${linker_flags}")
+         target_link_libraries(${TARGET_NAME} PUBLIC ${linker_flags})
+      else()
+         target_link_libraries(${TARGET_NAME} PUBLIC "-Wl,--whole-archive" ${ALL_LIBS} "-Wl,--no-whole-archive")
+      endif()
+      
+      add_dependencies(${TARGET_NAME} ${ALL_LIBS})
```
Comment (noseglasses) [inline cmake/Platform/Core/Targets/ArduinoFirmwareTargetCreator.cmake:55]: This explicit dependency specification is only necessary when supplying the linked libraries together with `-Wl,...` directives. In this case CMake seems unable to determine the dependency on libraries that are actually build as separate CMake targets.

**Review by MrPointer — APPROVED** [review]
Excellent as usual, thanks!

**Pull request merged**